src/main/java/com/mycompany/myapp/service/CSV/CSVParser.java

```diff
@@ -61,17 +66,13 @@
             });
     }
 
-    private static Boolean verify(List<ParsingContainerDTO> parsedObjs, List<String> poolIdInDatabase) {
-        Map<String, ParsingContainerDTO> objects =
-            parsedObjs.toLinkedMap(parsingContainerDTO -> Tuple.of(parsingContainerDTO.getPoolId(),
-                parsingContainerDTO));
-        return poolIdInDatabase
-            .forAll(poolId -> !objects.containsKey(poolId));
+    private static Boolean verify(List<ParsingContainerDTO> parsedObjs) {
+        return parsedObjs.toJavaStream().allMatch(object -> object.getMaximumCount() >= reservationService.getActiveOrInFutureReservedCount(object.getPoolId()));
```

Comment:
It is a vavr list so you don't need to call `toJavaStream`, instead simply use `forAll` on that list

Reply:
fixed